### PR TITLE
lzwolf: init at 20220104

### PR DESCRIPTION
**Pull request by tgunnoe**
###### Motivation for this change
No Wolfenstein 3D source port exists in nixpkgs

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -9469,6 +9469,12 @@
     githubId = 378734;
     name = "TG ⊗ Θ";
   };
+  tgunnoe = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:9472]: Please put this into a separate commit with the message ``maintainers: add tgunnoe``.

File: pkgs/games/lzwolf/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  buildInputs = [
+    SDL2 bzip2 zlib libjpeg SDL2_mixer SDL2_net libsndfile mpg123
+  ];
+
+  cmakeFlags = [
+    "-DCMAKE_BUILD_TYPE=Release"
+    "-DGPL=ON"
+  ];
+
+  enableParallelBuilding = true;
```
Comment (SuperSandro2000) [inline pkgs/games/lzwolf/default.nix:26]:
```suggestion
```

set by cmake automatically.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (tgunnoe) [reply maintainers/maintainer-list.nix:9472]: Done, thanks!

File: pkgs/games/lzwolf/default.nix (hunk 2 above)
Comment (tgunnoe) [reply pkgs/games/lzwolf/default.nix]: done, thanks!

**Comment by tgunnoe** [comment]
Hi. I made this pull request almost a year ago, before there were any Wolfenstein 3D source ports in nixpkgs at all. It was never merged even though I corrected the review.  

However, an inferior/outdated source port of Wolfenstein 3D was almost immediately merged by one of the "known" nix community members almost immediately, months later https://github.com/NixOS/nixpkgs/pull/131593.

This is why the nix project has a bottleneck. Nobody pays attention to the new contributors.

File: pkgs/games/lzwolf/default.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
+{ stdenv, lib, fetchFromBitbucket, p7zip, cmake
+, SDL2, bzip2, zlib, libjpeg
+, libsndfile, mpg123
+, SDL2_net, SDL2_mixer }:
+
+stdenv.mkDerivation rec {
+  pname = "lzwolf";
+  version = "20220104";
```
Comment (SuperSandro2000) [inline pkgs/games/lzwolf/default.nix:8]:
```suggestion
  version = "unstable-2022-01-04";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by tgunnoe** [comment]
Thank you @SuperSandro2000 ! I appreciate it
